### PR TITLE
Naive experimental way of trying to move packets (not perfect)

### PR DESCRIPTION
**Pull request by NybyDK**
NOT THE FINAL IMPLEMENTATION
THIS IS EXPERIMENTAL AND MEANT AS PROOF OF CONCEPT TO GET FEEDBACK

### CODE REVIEW

**Comment by NybyDK** [comment]
I will not merge until @TSKsmiley has reviewed, as he knows the most about TypeScript and the overall structure needs to be good
But @JakobTopholt and @Stormlighter feel free to review as well

File: src/lib/classes/MPLS/LER.ts [hunk 1]
```diff
@@ -49,6 +52,40 @@ export default class LER extends Router {
...
+      const nextHop = this.LIB.get(packet.label)?.nextHop;
+      if (!nextHop) return;
+
+      const newLabel = this.LIB.get(packet.label)?.outgoingLabel;
+      if (!newLabel) return;
+      packet.label = newLabel;
+      packet.node = { x: this.node.x, y: this.node.y };
+      const nextRouter = network.routers.find(
+        (router) => router instanceof CE && router.address === nextHop,
+      );
```
Comment (TSKsmiley) [inline src/lib/classes/MPLS/LER.ts:82]: we might want to map the routers by address and have a map per type

File: src/lib/classes/MPLS/Packet.ts [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+  public ttl: number = 32;
+  public nextHop: Router;
+
+  constructor(
+    public readonly id: number,
+    public readonly source: CE,
+    public readonly destination: CE,
+    public node: { x: number; y: number },
+  ) {
+    this.nextHop = paths.getPath(source.id, destination.id)?.[1] as Router;
```
Comment (TSKsmiley) [inline src/lib/classes/MPLS/Packet.ts:17]: i think we can use a type guard here as to not use as

File: src/lib/classes/NetworkStore.ts [hunk 3]
```diff
@@ -1,19 +1,21 @@
...
-  private store = writable<NetworkState>({ routers: [], connections: [] });
+  private store = writable<NetworkState>({ routers: [], connections: [], packets: [] });
 
   set = this.store.set;
   update = this.store.update;
   subscribe = this.store.subscribe;
 
   private _routers: Router[] = [];
   private _connections: Connection[] = [];
+  private _packets: Packet[] = [];
```
Comment (TSKsmiley) [inline src/lib/classes/NetworkStore.ts:18]: map the packets

**Review by TSKsmiley — CHANGES_REQUESTED** [review]

**Review by TSKsmiley — APPROVED** [review]

**Review by Stormlighter — DISMISSED** [review]

**Review by TSKsmiley — APPROVED** [review]

**Pull request merged**